core/src/main/java/com/expedia/www/haystack/client/Tracer.java

```diff
@@ -232,10 +248,20 @@ public Span start() {
         private ActiveSpanSource activeSpanSource = new ThreadLocalActiveSpanSource();
         private Clock clock = new SystemClock();
         private Dispatcher dispatcher;
+        private PropagationRegistry registry = new PropagationRegistry();
 
         public Builder(String serviceName, Dispatcher dispatcher) {
             this.serviceName = serviceName;
             this.dispatcher = dispatcher;
+
+            TextMapPropagator textMapPropagator = new TextMapPropagator.Builder().build();
```

Comment:
I don't understand this code...I see build() called, followed by withFormat() calls whose results are ignored. The typical builder pattern is "new Builder().withSomething(...).withSomethingElse(...).build(). What am I missing?

Reply:
`TextMapPropagator.Builder` vs. `Tracer.Builder`

Reply:
Got it. You might want to extract that code into a method named something like "setDefaultFormats()" to make that clear. After reading (part of) "clean code" I'm trying to move that direction myself.